backend/stakepoold/rpcclient.go

```diff
@@ -110,70 +113,201 @@ func connectWalletRPC(cfg *config) (*dcrrpcclient.Client, semver, error) {
 	return dcrwClient, walletVer, nil
 }
 
-func walletFetchUserTickets(ctx *appContext) (map[chainhash.Hash]string, string) {
+// nodeCheckTicketFee evaluates a stake pool ticket to see if it's
+// acceptable to the stake pool. The ticket must pay out to the stake
+// pool cold wallet, and must have a sufficient fee.
+// This function is a port of evaluateStakePoolTicket in dcrwallet which plucks
+// the in/out amounts from the RPC response rather than the wallet's record.
+// Stake pool fee tests are present in dcrwallet/wallet/txrules.
+func nodeCheckTicketFee(ctx *appContext, txhex string, blockhash string) (bool, error) {
+	// Create raw transaction.
+	var buf []byte
+	buf, err := hex.DecodeString(txhex)
+	if err != nil {
+		return false, fmt.Errorf("DecodeString failed: %v", err)
+	}
+	res, err := ctx.nodeConnection.DecodeRawTransaction(buf)
+	if err != nil {
+		return false, fmt.Errorf("DecodeRawTransaction failed: %v", err)
+	}
+
+	bhash, err := chainhash.NewHashFromStr(blockhash)
+	if err != nil {
+		return false, fmt.Errorf("NewHashFromStr failed for %v: %v", blockhash, err)
+	}
+	gbh, err := ctx.nodeConnection.GetBlockHeader(bhash)
+	if err != nil {
+		return false, fmt.Errorf("GetBlockHeader failed for %v: %v", bhash, err)
+	}
+
+	// Check the first commitment output (Vout[1])
+	// and ensure that the address found there exists
+	// in the list of approved addresses. Also ensure
+	// that the fee exists and is of the amount
+	// requested by the pool.
+
+	commitAmt := dcrutil.Amount(0)
+	feeAddr := ""
+	feeAddrValid := false
+	ticketValue, err := dcrutil.NewAmount(res.Vout[0].Value)
+	if err != nil {
+		return false, fmt.Errorf("NewAmount failed: %v", err)
+	}
+
+	for i := range res.Vout {
+		switch res.Vout[i].ScriptPubKey.Type {
+		case "sstxcommitment":
+			for j := range res.Vout[i].ScriptPubKey.Addresses {
+				_, exists := ctx.feeAddrs[res.Vout[i].ScriptPubKey.Addresses[j]]
+				if !exists {
+					continue
+				}
+				feeAddr = res.Vout[i].ScriptPubKey.Addresses[j]
+				feeAddrValid = true
+				feeAmount, err := dcrutil.NewAmount(*res.Vout[i].ScriptPubKey.CommitAmt)
+				if err != nil {
+					return false, fmt.Errorf("NewAmount failed: %v", err)
+				}
+				commitAmt = feeAmount
```

Comment:
I think this should break out of the for loop here, otherwise this will get the last fee address for a sstxcommitment output type (not that someone would really have 2 fee addresses at this point).  As your comment says it should be vout 1